bm_scrapy/middlewares.py

```diff
@@ -1,19 +1,28 @@
+import os
+
+from bm_scrapy.utils import parse_time
+from bm_scrapy.producer import connect_kafka_producer
 from scrapy.utils.request import request_fingerprint
-from bm_scrapy.writer import pipe_writer
 
 
 class StorageDownloaderMiddleware():
     def __init__(self):
-        self.writer = pipe_writer
+        self.producer = connect_kafka_producer()
 
     def process_response(self, request, response, spider):
-        self.writer.write_request(
-            url=response.url,
-            status=response.status,
-            method=request.method,
-            duration=request.meta.get('download_latency', 0) * 1000,
-            rsize=len(response.body),
-            fp=request_fingerprint(request),
-        )
+        data = {
+            'jid': os.getenv('BM_JOB'),
+            'payload': {
+                'url': response.url,
+                'status': int(response.status),
+                'method': request.method,
+                'duration': int(request.meta.get('download_latency', 0) * 1000),
+                'time': parse_time(),
+                'response_size': len(response.body),
+                'fingerprint': request_fingerprint(request),
+            },
+        }
+        self.producer.send('job_requests', value=data)
+        self.producer.flush()
```

Comment:
@rodp63  not so sure but `flush` is for sync call, maybe we should start to do send data asynchronously in this or other ticket

Reply:
Yes, it will be approached in: https://tasks.bitmaker.dev/issues/880